ES: Make it fail on unsupported installed IOSes

sepalani opened the pull request:
This PR is a response to: https://github.com/dolphin-emu/dolphin/pull/5225

Rather than handling IOSes that Dolphin doesn't emulate, let's pretend that they don't exist. Otherwise, it creates a weird behaviour where libogc believes that the context is acceptable for running functions like ```ES_GetNumTicketViews``` and ```ES_GetTicketViews```.

This PR fixes homebrew not booting while relying on IOS reloading of an (custom) IOS that isn't emulated by Dolphin but present in the NAND. These homebrew were blocked into an infinite loop during the ```IOS_ReloadIOS``` call.

I did some hardware tests too. If the app files are made empty or deleted, ditto for the TMD or the data/content is deleted, calling ```IOS_ReloadIOS``` will result in a black screen (no trace, no stack dump message or whatsoever). I assume, the console didn't like that at all. It was tested on a 3.2E Wii with a simple homebrew which calls ```IOS_ReloadIOS```.

Ready to be reviewed.

leoetlino commented on Source/Core/Core/IOS/ES/Views.cpp:
```diff
@@ -74,7 +89,12 @@ IPCCommandResult ES::GetTicketViews(const IOCtlVRequest& request)
+  if (IsUnsupportedIOS(TitleID))
+  {
+    ERROR_LOG(IOS_ES, "GetViews: Dolphin doesn't emulate IOS title %016" PRIx64, TitleID);
+    return GetDefaultReply(ES_EINVAL);
```
If the intent is to make it look like there is no ticket for the IOS title, then this should be removed. When no ticket is found, ES doesn't return any particular error code.

BhaaLseN replied:
It has to return something, right? Or does it simply yield control and whatever garbage was left in the return register ends up with the caller?

leoetlino replied:
Yep, as far as I can see it just doesn't touch the ioctlv output buffer and returns IPC_SUCCESS.

sepalani commented:
@leoetlino
As explained in the previous PR, a timing issue was making Dolphin's reply overwritten. Otherwise, it does check ```ES_Launch``` return value. Regardless, I addressed your comment and added another check in ```ESLaunch``` since it does succeed (```GetNoReply()```) for unsupported installed IOSes.

You'll find in attachment the result of my hardware tests and the source code (based on libogc) + ELF.
[hardware_ios_reload.zip](https://github.com/dolphin-emu/dolphin/files/1714652/hardware_ios_reload.zip)

![ios_reload_36](https://user-images.githubusercontent.com/7890055/36076516-5b401a30-0f5d-11e8-9dd7-61faf4f82f88.jpg)
![ios_reload_121](https://user-images.githubusercontent.com/7890055/36076517-5b95f838-0f5d-11e8-86e1-8544d73d31f0.jpg)
![ios_reload_249](https://user-images.githubusercontent.com/7890055/36076518-5bea0d88-0f5d-11e8-9f42-94e75550b676.jpg)
